template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java

```diff
@@ -59,39 +59,45 @@ public class CoreConfigProvider extends AbstractRoleConfigProvider {
-                    .ifPresent(location -> apiClusterTemplateConfigs.add(config(CORE_DEFAULTFS, location.getValue())));
+        if (!templateProcessor.isRoleTypePresentInService(HDFS, Lists.newArrayList(NAMENODE))
+                && source.getFileSystemConfigurationView().isPresent()
+                && ConfigUtils.getStorageLocationForServiceProperty(source, CORE_DEFAULTFS).isPresent()) {
+            Optional<ApiClusterTemplateConfig> roleConfig = templateProcessor.getRoleConfig(CORE_SETTINGS, STORAGEOPERATIONS, CORE_DEFAULTFS);
+            if (roleConfig.isEmpty()) {
+                ConfigUtils.getStorageLocationForServiceProperty(source, CORE_DEFAULTFS)
+                        .ifPresent(location -> apiClusterTemplateConfigs.add(config(CORE_DEFAULTFS, location.getValue())));
+            }
+
+            StringBuilder hdfsCoreSiteSafetyValveValue = new StringBuilder();
+            s3ConfigProvider.getServiceConfigs(source, hdfsCoreSiteSafetyValveValue);
+            adlsConfigProvider.populateServiceConfigs(source, hdfsCoreSiteSafetyValveValue, templateProcessor.getStackVersion());
+
+            hdfsCoreSiteSafetyValveValue.append(ConfigUtils.getSafetyValveProperty(HADOOP_SECURITY_GROUPS_CACHE_BACKGROUND_RELOAD, "true"));
+
+            if (!hdfsCoreSiteSafetyValveValue.toString().isEmpty()) {
+                LOGGER.info("Adding '{}' to the cluster template config.", CORE_SITE_SAFETY_VALVE);
+                apiClusterTemplateConfigs.add(config(CORE_SITE_SAFETY_VALVE, hdfsCoreSiteSafetyValveValue.toString()));
+            }
         }
-
-        StringBuilder hdfsCoreSiteSafetyValveValue = new StringBuilder();
-        s3ConfigProvider.getServiceConfigs(source, hdfsCoreSiteSafetyValveValue);
-        adlsConfigProvider.populateServiceConfigs(source, hdfsCoreSiteSafetyValveValue, templateProcessor.getStackVersion());
-
-        hdfsCoreSiteSafetyValveValue.append(ConfigUtils.getSafetyValveProperty(HADOOP_SECURITY_GROUPS_CACHE_BACKGROUND_RELOAD, "true"));
-
-        if (!hdfsCoreSiteSafetyValveValue.toString().isEmpty()) {
-            LOGGER.info("Adding '{}' to the cluster template config.", CORE_SITE_SAFETY_VALVE);
-            apiClusterTemplateConfigs.add(config(CORE_SITE_SAFETY_VALVE, hdfsCoreSiteSafetyValveValue.toString()));
+        if (entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
+                && source.getStackType().equals(StackType.DATALAKE)) {
+            apiClusterTemplateConfigs.add(config(HADOOP_RPC_PROTECTION, "privacy"));
         }
 
         return apiClusterTemplateConfigs;
     }
 
     @Override
     public Map<String, ApiClusterTemplateService> getAdditionalServices(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
-        if (isConfigurationNeeded(cmTemplateProcessor, source)
+        if (!cmTemplateProcessor.isRoleTypePresentInService(HDFS, Lists.newArrayList(NAMENODE))
```

Comment:
This if condition and the previous one on L68 is the same. Please add a meaningful name to it an make a helper method.